lib/index.js

```diff
@@ -126,8 +128,25 @@ api.set = async ({
   } else if(type === 'challenge') {
     // TODO: implement
     throw new Error('Not implemented.');
+  } else if(type === 'totp') {
+    // check for existing totp secret
+    let err;
+    try {
+      await api.get({actor: null, account, email, type});
+    } catch(e) {
+      if(e.name !== 'NotFoundError') {
+        throw e;
+      }
+      err = e;
+    }
+    // a NotFoundError is expected here
+    if(!err) {
+      // FIXME: if there is an existing token for totp, what is the error
+      // message?
+      throw new Error('THIS IS NOT ALLOWED, RIGHT?');
```

Comment:
I wonder if we just want to overwrite rather than require an explicit removal. If we want to allow overwrite -- no error, if we want to require removal first, then this would be a `DuplicateError` "Time-based one time password authentication token already set." or something like that.

Reply:
Throwing error for now.